x/emissions/keeper/inference_synthesis/network_inference_builder.go

```diff
@@ -107,6 +107,150 @@ func getForecastImpliedInferences(
 	return forecastImpliedInferences
 }
 
+type GetOneOutInfererForecastImpliedInferencesArgs struct {
+	Ctx                  sdk.Context
+	K                    emissionskeeper.Keeper
+	Logger               log.Logger
+	TopicId              uint64
+	Inferers             []Inferer
+	InfererToInference   map[Inferer]*emissions.Inference
+	InfererToRegret      map[Inferer]*Regret
+	AllInferersAreNew    bool
+	Forecasters          []Forecaster
+	ForecasterToForecast map[Forecaster]*emissions.Forecast
+	ForecasterToRegret   map[Forecaster]*Regret
+	NetworkCombinedLoss  alloraMath.Dec
+	EpsilonTopic         alloraMath.Dec
+	PNorm                alloraMath.Dec
+	CNorm                alloraMath.Dec
+	StdDevPlusEpsilon    alloraMath.Dec
+}
+
+// GetOneOutInfererForecastImpliedInferences calculates what each forecaster's implied inference
+// would be when each inferer is removed from the calculation
+func GetOneOutInfererForecastImpliedInferences(args GetOneOutInfererForecastImpliedInferencesArgs) (
+	oneOutInfererForecastImpliedValues []*emissions.OneOutInfererForecasterValues,
+	err error,
+) {
+	oneOutInfererForecastImpliedValues = make([]*emissions.OneOutInfererForecasterValues, 0)
+
+	// Only process if we have both inferers and forecasters
+	if len(args.Inferers) <= 1 || len(args.Forecasters) == 0 {
+		return oneOutInfererForecastImpliedValues, nil
+	}
+
+	// Organize by forecaster first, then by withheld inferer
+	for _, forecaster := range args.Forecasters {
+		// For each forecaster, we'll calculate what their forecast-implied inference would be
+		// if each inferer was removed one at a time
+		oneOutInfererValues := make([]*emissions.WithheldWorkerAttributedValue, 0)
+
+		for _, withheldInferer := range args.Inferers {
+			// Filter out the inferer we want to withhold
+			filteredInferers := make([]Inferer, 0, len(args.Inferers)-1)
+			filteredInfererToInference := make(map[Inferer]*emissions.Inference, len(args.InfererToInference)-1)
+			filteredInfererToRegret := make(map[Inferer]*Regret, len(args.InfererToRegret)-1)
+
+			for _, inferer := range args.Inferers {
+				if inferer != withheldInferer {
+					filteredInferers = append(filteredInferers, inferer)
+					if inference, ok := args.InfererToInference[inferer]; ok {
+						filteredInfererToInference[inferer] = inference
+					}
+					if regret, ok := args.InfererToRegret[inferer]; ok {
+						filteredInfererToRegret[inferer] = regret
+					}
+				}
+			}
+
+			// Get this forecaster's forecast and filter out the withheld inferer
+			forecast, ok := args.ForecasterToForecast[forecaster]
+			if !ok {
+				continue
+			}
+
+			filteredForecastElements := make([]*emissions.ForecastElement, 0)
+			for _, element := range forecast.ForecastElements {
+				if element.Inferer != withheldInferer {
+					filteredForecastElements = append(filteredForecastElements, element)
+				}
+			}
```

Comment:
```suggestion
			filteredForecastElements = filteredForecastElements[:0]
			copy(filteredForecastElements, forecast.ForecastElements)
			if i := slices.IndexFunc(forecast.ForecastElements, func(element *emissions.ForecastElement) bool {
				return element.Inferer == withheldInferer
			}); i != -1 {
				filteredForecastElements[i] = filteredForecastElements[0]
				filteredForecastElements = filteredForecastElements[1:]
			}
```

Performance improvement as well, the `filteredForecastElements ` should be allocated at the beginning of the func